### PR TITLE
Create valgrind.adoc

### PR DESCRIPTION
**Pull request by paulfloyd**
Valgrind news.

### CODE REVIEW

File: website/content/en/status/report-2024-01-2024-03/valgrind.adoc [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+Support for FreeBSD 13.3 has been added.
+Added a redirect for `reallocarray`.
+Several fixes for `aio*` functions.
+Added a redirect for `memccpy`.
+There is a fix for `_umtx_op OP_ROBUST_LISTS`.
+Added redirs for C23 `free_sized` and `free_aligned_sized`.
+Correctly propagate the ELF stack protection flags to the guest stack that Valgrind synthesizes.
+Fixes for `--sanity-level-3` and above (only used for Valgrind self-testing at runtime).
+Aeveral fixes to checking done for `semctl`.
```
Comment (pauamma) [inline website/content/en/status/report-2024-01-2024-03/valgrind.adoc:32]: s/Aeveral/Several/

File: website/content/en/status/report-2024-01-2024-03/valgrind.adoc [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+Fixed argument checking for `utrace`.
```
[the remark this answers is not in the document]
Comment (paulfloyd) [reply website/content/en/status/report-2024-01-2024-03/valgrind.adoc:33]: That would work.

File: website/content/en/status/report-2024-01-2024-03/valgrind.adoc [hunk 3]
```diff
@@ -19,15 +19,16 @@ Handling of closefrom has been improved to use this feature.
...
-Aeveral fixes to checking done for `semctl`.
-Fixed argument checking for `utrace`.
+Here is a list of smaller bugfixes:
+
+* Support for FreeBSD 13.3 has been added.
+* Added a redirect for `reallocarray`.
+* Several fixes for `aio*` functions.
+* Added a redirect for `memccpy`.
+* There is a fix for `_umtx_op OP_ROBUST_LISTS`.
+* Added redirs for C23 `free_sized` and `free_aligned_sized`.
```
Comment (pauamma) [inline website/content/en/status/report-2024-01-2024-03/valgrind.adoc:29]:
```suggestion
* Added redirects for C23 `free_sized` and `free_aligned_sized`.
```

**Review by pauamma — APPROVED** [review]
With these 2 nits (fixable on commit).

File: website/content/en/status/report-2024-01-2024-03/valgrind.adoc (hunk 3 above)
Comment (paulfloyd) [reply website/content/en/status/report-2024-01-2024-03/valgrind.adoc:29]: OK

File: website/content/en/status/report-2024-01-2024-03/valgrind.adoc [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+Here is a list of smaller bugfixes:
+
+* Support for FreeBSD 13.3 has been added.
+* Added a redirect for `reallocarray`.
+* Several fixes for `aio*` functions.
+* Added a redirect for `memccpy`.
+* There is a fix for `_umtx_op OP_ROBUST_LISTS`.
+* Added redirs for C23 `free_sized` and `free_aligned_sized`.
+* Correctly propagate the ELF stack protection flags to the guest stack that Valgrind synthesizes.
```
[the remark this answers is not in the document]
Comment (paulfloyd) [reply website/content/en/status/report-2024-01-2024-03/valgrind.adoc:30]: No, it should be present continuous, this is something that gets done at every startup. Previously this was hard coded and did not respect the guest stack protection flags.

File: website/content/en/status/report-2024-01-2024-03/valgrind.adoc [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+link:https://www.valgrind.org/[Valgrind Home Page] URL: link:https://www.valgrind.org/[] +
+link:https://www.valgrind.org/docs/manual/dist.news.html[Valgrind News] URL: link:https://www.valgrind.org/docs/manual/dist.news.html[]
+link:https://github.com/paulfloyd/freebsdarm64_valgrind[arm64 port] URL: link:https://github.com/paulfloyd/freebsdarm64_valgrind[]
+
+Contact: Paul Floyd <pjfloyd@wanadoo.fr>
+
+The major news, as per the title, is that a port to FreeBSD arm64 (or aarch64) is now ready.
+The next steps are to get it reviewed and pushed upstream.
+
+Valgrind 3.23 due out at the end of April 2024 and package:devel/valgrind[] will be updated shortly after that.
```
Comment (lsalvadore) [inline website/content/en/status/report-2024-01-2024-03/valgrind.adoc:13]: Mind that I am not a native English speaker, but this does not sound right to me. Maybe
```suggestion
Valgrind 3.23 is due out at the end of April 2024 and package:devel/valgrind[] will be updated shortly after that.
```

File: website/content/en/status/report-2024-01-2024-03/valgrind.adoc [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+package:devel/valgrind-devel[] will get an update as soon as I've pushed the changes for arm64.
```
Comment (lsalvadore) [inline website/content/en/status/report-2024-01-2024-03/valgrind.adoc:15]: Please avoid contractions.
```suggestion
package:devel/valgrind-devel[] will get an update as soon as I have pushed the changes for arm64.
```

**Review by lsalvadore — COMMENTED** [review]
Thanks for your report. I made only two small comments, I can fix those on commit if your prefer.

File: website/content/en/status/report-2024-01-2024-03/valgrind.adoc (hunk 5 above)
Comment (paulfloyd) [reply website/content/en/status/report-2024-01-2024-03/valgrind.adoc:13]: OK

File: website/content/en/status/report-2024-01-2024-03/valgrind.adoc (hunk 6 above)
Comment (paulfloyd) [reply website/content/en/status/report-2024-01-2024-03/valgrind.adoc:15]: OK as well

**Comment by lsalvadore** [comment]
Merged with the approved small changes. Thanks!

**Pull request closed without merging**